File: process.php
<?php

echo "Running!\n";

if ( !array_key_exists( 1, $argv ) ) {
	throw new Exception( 'Not passed a log file.' );
}
$logFileLocation = $argv[1];
if ( !file_exists( $logFileLocation ) || is_readable( $logFileLocation ) ) {
	throw new Exception( 'Can\'t read log file: ' . $logFileLocation );
}

$outputDirectory = __DIR__ . DIRECTORY_SEPARATOR . 'output';
if ( !is_dir( $outputDirectory ) ) {
	mkdir( $outputDirectory );
}
$outputFileLocation = $outputDirectory . DIRECTORY_SEPARATOR . date( "Y-m-d_H:i:s" ) . '.csv';
$outputHandle = fopen( $outputFileLocation , 'w' );
if ( !$outputHandle ) {
	throw new Exception( 'Could not get output handle' );
}

$inputHandle = fopen( $logFileLocation, "r" );

echo "Input from: $logFileLocation\n";
echo "Output to: $outputFileLocation\n";

if ( $inputHandle ) {
	$headers = array(
		'timestamp', 'oldid', 'newid', 'oldtimestamp', 'newtimestamp', 'pageid', 'revisions',
		'intermediate', 'olderrevs', 'newerrevs', 'revslider'
	);
	sort( $headers );
	fputcsv( $outputHandle, $headers );
	while ( ( $line = fgets( $inputHandle ) ) !== false ) {
		$lineParts = explode( ' dewiki_diffstats DEBUG: dewiki diff page view ', $line );
		// Account for some logs missing some keys (such as revslider) by auto filling them as blank
		$data = array_map(function() { return ''; }, array_flip( $headers ));
		$data = array_merge($data, json_decode( trim( $lineParts[1] ), true ));
		ksort( $data );
		fputcsv( $outputHandle, $data );
	}

	fclose( $inputHandle );
} else {
	throw new Exception( 'Could not get input handle.' );
}

fclose( $outputHandle );

echo "Done!\n";

File: process.py
import sys
import os.path
import time
import csv
import json

print "Running!"

if len(sys.argv) != 2:
    raise Exception('Not passed a log file.')
logFileLocation = sys.argv[1]
if os.path.isfile(logFileLocation) == False and os.access(logFileLocation, os.R_OK) == False:
    raise Exception('Can\'t read log file: ' + logFileLocation)

outputDirectory = os.path.join(os.path.dirname(os.path.realpath(__file__)),"output")
if not os.path.exists(outputDirectory):
    os.makedirs(outputDirectory)
outputFileLocation = os.path.join(outputDirectory,time.strftime('%Y-%m-%d_%H:%M:%S') + '.csv')

print "Input from: " + logFileLocation
print "Output to: " + outputFileLocation

with open(outputFileLocation, 'w') as outputHandle:
    headers = ['timestamp', 'oldid', 'newid', 'oldtimestamp', 'newtimestamp', 'pageid', 'revisions',
               'intermediate', 'olderrevs', 'newerrevs', 'revslider']
    headers.sort()
    writer = csv.DictWriter(outputHandle, fieldnames=headers)
    writer.writeheader()
    with open(logFileLocation) as inputHandle:
        for line in inputHandle:
            lineParts = line.split(' dewiki_diffstats DEBUG: dewiki diff page view ')
            data = json.loads(lineParts[1])
            writer.writerow(data)

print "Done!"
